core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -421,7 +423,10 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
     else Some(Serialization.Information(address, context.system))
   }
 
-  private def serializeEvent(p: PersistentRepr, tags: Set[String]): Serialized = {
+  protected def serializeEvent(p: PersistentRepr, tags: Set[String]): Serialized = {
+    // use same clock source as the UUID for the timeBucket
+    val nowUuid = UUIDs.timeBased()
```

Comment:
here is the UUID that I mentioned in previous comments, since this is done in futures we don't know that they are monotonically increasing when they arrive in the tag writer. would be good to move the creation of the UUIDs into the actor, before serialization

Reply:
Even if we move them to the actor they could over take each other as they are sent to the tag writer on completion of a cassandra write. So I think it is best to just have the tag writer not assume they'll be in order, @patriknw WDYT?

Reply:
👍

Reply:
This is done now so I didn't change the Journal to create the timeuuids in the actor.